extensions-core/druid-basic-security/src/test/java/org/apache/druid/security/basic/authentication/endpoint/BasicAuthenticatorResourceTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.security.basic.authentication.endpoint;
+
+import org.apache.druid.security.basic.authentication.entity.BasicAuthenticatorCredentialUpdate;
+import org.apache.druid.server.security.AuthValidator;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.servlet.http.HttpServletRequest;
+import java.nio.charset.StandardCharsets;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BasicAuthenticatorResourceTest
+{
+  private static final String AUTHENTICATOR_NAME = "AUTHENTICATOR_NAME";
+  private static final String INVALID_AUTHENTICATOR_NAME = "INVALID_AUTHENTICATOR_NAME";
+  private static final String USER_NAME = "USER_NAME";
+  private static final byte[] SERIALIZED_USER_MAP = "SERIALIZED_USER_MAP".getBytes(StandardCharsets.UTF_8);
+  @Mock(answer = Answers.RETURNS_MOCKS)
+  private BasicAuthenticatorResourceHandler handler;
+  @Mock
+  private AuthValidator authValidator;
+  @Mock
+  private HttpServletRequest req;
+  @Mock
+  private BasicAuthenticatorCredentialUpdate update;
+
+  private BasicAuthenticatorResource target;
+
+  @Before
+  public void setUp()
+  {
+    Mockito.doThrow(IllegalArgumentException.class)
+           .when(authValidator)
+           .validateAuthenticatorName(INVALID_AUTHENTICATOR_NAME);
```

Comment:
Why do we mock `authValidator` here? It doesn't seem hard to use the real method. I would suggest to use the real method in that case since we can catch potential side effects easily if we modify the logic of `authValidator`.

Reply:
Same for `BasicAuthorizerResourceFilter`.

Reply:
I didn't want to include the logic in this unit test, that's why I chose to mock it. I like this de-coupling because if the logic in AuthValidator changes in the future, we don't need to update all the test classes that use AuthValidator.

The tests that validate whether or not a name is valid are written in `AuthValidatorTest`, so any change in logic should be caught by that test. 

These tests just validate that if the AuthValidator encounters an invalid auth name, an appropriate exception is thrown.

Reply:
OK.